search/searcher/search_geopolygon.go

```diff
@@ -16,34 +16,51 @@ package searcher
 
 import (
 	"fmt"
+	"math"
+
 	"github.com/blevesearch/bleve/v2/geo"
 	"github.com/blevesearch/bleve/v2/numeric"
 	"github.com/blevesearch/bleve/v2/search"
 	index "github.com/blevesearch/bleve_index_api"
-	"math"
 )
 
 func NewGeoBoundedPolygonSearcher(indexReader index.IndexReader,
-	polygon []geo.Point, field string, boost float64,
+	coordinates []geo.Point, field string, boost float64,
 	options search.SearcherOptions) (search.Searcher, error) {
-
-	if len(polygon) < 3 {
+	if len(coordinates) < 3 {
 		return nil, fmt.Errorf("Too few points specified for the polygon boundary")
 	}
 
-	// compute the bounding box enclosing the polygon
-	topLeftLon, topLeftLat, bottomRightLon, bottomRightLat, err :=
-		geo.BoundingRectangleForPolygon(polygon)
-	if err != nil {
-		return nil, err
+	var rectSearcher search.Searcher
```

Comment:
Recommend a rename here, maybe `polySearcher`

Reply:
It's not a polygon searcher, rather a bounded rectangle searcher.

Reply:
Oh i see - we use a bounding box to enclose a polygon. How did precision work here?

Reply:
nothing changed here. All the geo queries have this filtering phase in the implementation.

Reply:
my lapse of memory :)